sdk/context.ts

```diff
@@ -0,0 +1,97 @@
+import { NetworkId } from '@synthetixio/contracts-interface';
+import { Provider as EthCallProvider } from 'ethcall';
+import { ethers } from 'ethers';
+
+import * as sdkErrors from './common/errors';
+import { ContractName, getContractsByNetwork, ContractMap } from './contracts';
+
+export interface IContext {
+	provider?: ethers.providers.Provider;
```

Comment:
Technically we can have cases where clients may want to lazily set the provider using the `setProvider` method after creating an instance. However, since we're only considering Kwenta's use case right now, we don't really need to worry about that.